streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamTransformTest.java

```diff
@@ -52,7 +82,13 @@ public void testTransform() {
                         private int total = 0;
 
                         @Override
-                        public void init(ProcessorContext context) {
+                        public void init(final ProcessorContext context) {
+                            context.schedule(1, PunctuationType.WALL_CLOCK_TIME, new Punctuator() {
```

Comment:
I am wondering, if we should rewrite this (I understand that we must if we want to us `TopologyTestDriver` that only supports the new punctuation API) because we should have tests for the old deprecated API as long as the API is not removed... \cc @guozhangwang @bbejeck @vvcephei WDYT?

Reply:
It's a +1 from me to update this test to cover the deprecated API.  IMHO, as long as it's still in the code base we shouldn't remove any tests covering the old API.

Reply:
Yes, it is wise to keep the test for the old API. I would keep the old test, which uses the old punctuation api with `KStreamTestDriver`, and add the above test which covers the new punctuation and `TopologyTestDriver`. Is that ok?

Reply:
Sounds right to me. It's probably deserving of a comment to explain the rationale, though.